Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/ArtifactKey.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Central Repository
+ *
+ * Copyright 2015-2017 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.centralrepository.contentviewer;
+
+import java.util.Objects;
+
+/**
+ * Used as a key to ensure we eliminate duplicates from the result set by not overwriting CR correlation instances.
+ */
+final class ArtifactKey {
+    
+    private final String dataSource;
```

Comment:
I think this would be more clear if you renamed this to dataSourceDeviceID (or similar) and updated  the name of getDataSource(). The constructor is a bit ambiguous as it stands.